ios/Podfile

```diff
@@ -0,0 +1,37 @@
+# This was the iOS Deployment Target (project > ZulipMobile > Info in Xcode)
+# as of 2020-03.
+platform :ios, '10.3'
+
+require_relative '../node_modules/react-native-unimodules/cocoapods.rb'
+
+# See
+# https://github.com/facebook/react-native-website/blob/ded79d2cf4456d8b1a4f67c2cdc1391789e70617/docs/integration-with-existing-apps.md#configuring-cocoapods-dependencies
+# for the choice of pods here.
+target 'ZulipMobile' do
+  pod 'React', :path => '../node_modules/react-native', :subspecs => [
+    'Core',
+    'CxxBridge', # Include this for RN >= 0.47
+    'DevSupport', # Include this to enable In-App Devmenu if RN >= 0.43
+    'RCTText',
+    'RCTNetwork',
+    'RCTWebSocket', # Needed for debugging
+    'RCTAnimation', # Needed for FlatList and animations running on native UI thread
+    'RCTImage', # Included in react-native-unimodules' example Podfile
+    # Add any other subspecs you want to use in your project
```

Comment:
> That is to name here all the subspecs we need
> [...]
> It looks like you can find all the available subspecs by looking in `node_modules/react-native/React.podspec`.

In the Podfile, should I aim to reference all of the subspecs that are listed in React.podspec? I took the example Podfile from the RN v0.59 doc thinking that was the minimum set of subspecs necessary, and that anything on top of that might be pointless or introduce bugs. I can see the argument for being more complete than we may need to be, but I wonder why the example Podfile doesn't include all the subspecs in React.podspec.

Do you think "all the subspecs in React.podspec" includes just the outer level of nesting, here, or might we also need to get the "subspecs of the subspecs" (if that makes sense):

`$ grep '\.subspec \"[a-zA-Z]*\" do' node_modules/react-native/React.podspec`
```
  s.subspec "Core" do |ss|
  s.subspec "CxxBridge" do |ss|
  s.subspec "DevSupport" do |ss|
  s.subspec "RCTFabric" do |ss|
  s.subspec "tvOS" do |ss|
  s.subspec "jsinspector" do |ss|
  s.subspec "jsiexecutor" do |ss|
  s.subspec "jsi" do |ss|
  s.subspec "PrivateDatabase" do |ss|
  s.subspec "cxxreact" do |ss|
  s.subspec "fabric" do |ss|
    ss.subspec "activityindicator" do |sss|
    ss.subspec "attributedstring" do |sss|
    ss.subspec "core" do |sss|
    ss.subspec "debug" do |sss|
    ss.subspec "graphics" do |sss|
    ss.subspec "scrollview" do |sss|
    ss.subspec "text" do |sss|
    ss.subspec "textlayoutmanager" do |sss|
    ss.subspec "uimanager" do |sss|
    ss.subspec "view" do |sss|
  s.subspec "RCTFabricSample" do |ss|
  s.subspec "ART" do |ss|
  s.subspec "RCTActionSheet" do |ss|
  s.subspec "RCTAnimation" do |ss|
  s.subspec "RCTBlob" do |ss|
  s.subspec "RCTCameraRoll" do |ss|
  s.subspec "RCTGeolocation" do |ss|
  s.subspec "RCTImage" do |ss|
  s.subspec "RCTNetwork" do |ss|
  s.subspec "RCTPushNotification" do |ss|
  s.subspec "RCTSettings" do |ss|
  s.subspec "RCTText" do |ss|
  s.subspec "RCTVibration" do |ss|
  s.subspec "RCTWebSocket" do |ss|
  s.subspec "fishhook" do |ss|
  s.subspec "RCTLinkingIOS" do |ss|
  s.subspec "RCTTest" do |ss|
```

Reply:
> Hmm, so, I think there's a next step to be done before we're really fully _using_ CocoaPods. That is to name here all the subspecs we need... and remove all the corresponding pieces from our own Xcode project file.

That's certainly something we're going to have to do at some point, but it may be more complexity than is needed for #3964. Does `react-native-unimodules` need a *full* CocoaPods conversion, or will a bare-bones Podfile without the React Native core components suffice? (The latter would probably still be a significant step towards #3983.)

Reply:
> Does `react-native-unimodules` need a _full_ CocoaPods conversion, or will a bare-bones Podfile without the React Native core components suffice? (The latter would probably still be a significant step towards #3983.)

Yeah, if the fuller conversion turns out to be more complex, I'd be happy merging whatever smaller version suffices to unblock #3964.

Though again, I'll be more confident that we've successfully fully wired up CocoaPods when there are pieces of the build that we rely on and don't have some other way they get included :wink: Without that, it seems like we might attempt #3964 and then discover there's something else we need to do in order for a library we add there to actually get built into the app.

Reply:
> In the Podfile, should I aim to reference all of the subspecs that are listed in React.podspec? I took the example Podfile from the RN v0.59 doc thinking that was the minimum set of subspecs necessary, and that anything on top of that might be pointless or introduce bugs. I can see the argument for being more complete than we may need to be, but I wonder why the example Podfile doesn't include all the subspecs in React.podspec

What I think I'd ideally like is to just specify the exact same set as we currently do in the Xcode project file, and at the same time stop specifying those there. I.e. just convert over *how* we include those pieces, separate from any changes to which ones we include.

It's also pretty likely that there are some things we include in the build that we don't need to, largely because the Xcode config has felt so opaque and it's felt so finicky, and liable to cause problems that are annoying to debug, to modify it. So after they're all in this Podfile instead, we might go through and identify some we don't need and remove those.

Reply:
> and remove all the corresponding pieces from our own Xcode project file

This sounds easier to solve than determining the correct contents of the Podfile. So far, I haven't found a clear way forward for how to cleanly remove things from the project.pbxproj file. I think this area of the Xcode UI will be helpful:

![image](https://user-images.githubusercontent.com/22248748/77216217-09e88380-6ad6-11ea-9eb2-9a02f51b2591.png)

Note that `libPods-ZulipMobile.a` is in that list, and a lot of other, mostly RN-related stuff is also in the list. It seems plausible to want to shrink that list down to just the Pods item, but I'd need a clean mapping from a given `.a` item in that list, to a corresponding added entry in the Podfile.

Reply:
Mm, our posts crossed! 🙂Still, to accomplish 

> specify the exact same set as we currently do in the Xcode project file

it sounds like the next step is for me to find a mapping from those `.a` files in that screenshot (assuming that screenshot controls the Xcode project file in the ways we want) to pods/subspecs that can be added to the Podfile.